coreax/score_matching.py

```diff
@@ -605,10 +540,36 @@ def score_function(x_):
         return score_function
 
 
-# Define the pytree node for the added class to ensure methods with JIT decorators
-# are able to run. This tuple must be updated when a new class object is defined.
-score_matching_classes = (SlicedScoreMatching, KernelDensityMatching)
-for _current_class in score_matching_classes:
-    tree_util.register_pytree_node(
-        _current_class, _current_class.tree_flatten, _current_class.tree_unflatten
-    )
+def convert_stein_kernel(
+    x: ArrayLike,
+    kernel: Kernel,
+    score_matching: Union[ScoreMatching, None],
+) -> SteinKernel:
+    r"""
+    Convert the kernel to a :class:`~coreax.kernel.SteinKernel`.
+
+    :param x: The data used to call `score_matching.match(x)`
+    :param kernel: :class:`~coreax.kernel.Kernel` instance implementing a kernel
+        function :math:`k: \mathbb{R}^d \times \mathbb{R}^d \rightarrow \mathbb{R}`;
+        if 'kernel' is a :class:`~coreax.kernel.SteinKernel` and :code:`score_matching
+        is not None`, a new instance of the kernel will be generated where the score
+        function is given by :code:`score_matching.match(x)`
+    :param score_matching: Specifies/overwrite the score function of the implied/passed
+       :class:`~coreax.kernel.SteinKernel`; if :data:`None`, default to
+       :class:`~coreax.score_matching.KernelDensityMatching` unless 'kernel' is a
+       :class:`~coreax.kernel.SteinKernel`, in which case the kernel's existing score
+       function is used.
+    :return: The (potentially) converted/updated :class:`~coreax.kernel.SteinKernel`.
+    """
+    if isinstance(kernel, SteinKernel):
+        if score_matching is not None:
+            _kernel = eqx.tree_at(
+                lambda x: x.score_function, kernel, score_matching.match(x)
+            )
+        _kernel = kernel
```

Comment:
Should this be `else: _kernel = kernel` instead? Currently this just overrides the value of `_kernel` that might have been set above. (Since `eqx.tree_at` doesn't modify anything in-place, the statement above doesn't change `kernel`).

Reply:
This was a straight copy-paste so I think this might have been a bug that was missed in a previous pull, fixed!